### PR TITLE
make it works in visual mode

### PR DESCRIPTION
**Pull request by zhaozg**
ugly, but works

nnoremap <localleader>h :call http#do_buffer(1, '')<CR>
vnoremap <localleader>h :<C-U>call http#do_buffer(1, "'<,'>")<CR>

### CODE REVIEW

**Comment by nicwest** [comment]
hey bud sorry totally missed this, let me check it out this weekend.

File: autoload/http.vim [hunk 1]
```diff
@@ -16,29 +16,48 @@ function! s:new_request() abort
...
-    
 endfunction
 
 let s:pre_clean_uri_line_pattern = '^\(\a*\) \(.*\) HTTP/\([0-9.]\+\)\|$'
 let s:uri_line_pattern = '^\(OPTIONS\|GET\|HEAD\|POST\|PUT\|DELETE\|TRACE\|CONNECT\|PATCH\) \(.*\) HTTP/\([0-9.]\+\)$'
 let s:header_line_pattern = '^\([^:]\+\): \(.*\)$'
 
-function! s:parse_request_buffer(buffer, pattern, follow) abort
+function! s:get_visual_selection()
+    " Why is this not a built-in Vim script function?!
```
Comment (nicwest) [inline autoload/http.vim]: tell me about it!

File: autoload/http.vim [hunk 2]
```diff
@@ -16,29 +16,48 @@ function! s:new_request() abort
...
+    let [line_start, column_start] = getpos("'<")[1:2]
+    let [line_end, column_end] = getpos("'>")[1:2]
+    let lines = getline(line_start, line_end)
+    if len(lines) == 0
+        return []
+    endif
+    let lines[-1] = lines[-1][: column_end - (&selection == 'inclusive' ? 1 : 2)]
+    let lines[0] = lines[0][column_start - 1:]
```
Comment (nicwest) [inline autoload/http.vim]: is column specific visual selection necessary here? if not then we can just use `-range` on the `:Http` command

File: autoload/http.vim [hunk 3]
```diff
@@ -16,29 +16,48 @@ function! s:new_request() abort
...
+function! s:get_lines(buffer, range)
+    if a:range=="'<,'>"
+      return s:get_visual_selection()
+    end
+    if a:range=='' || a:range=='%'
+      return getbufline(a:buffer, 0, '$')
+    end
+endfunction
+
+function! s:parse_request_buffer(lines, pattern, follow) abort
```
Comment (nicwest) [inline autoload/http.vim]: This very specifically doesn't do anything with buffer any more!

```suggestion
function! s:parse_request(lines, pattern, follow) abort
```

**Comment by nicwest** [comment]
I like this a lot! shout if you don't have time to finish this up and I will happily do so.

**Comment by zhaozg** [comment]
> I like this a lot! shout if you don't have time to finish this up and I will happily do so.

please continues, I'm not good at vim-script

**Comment by nicwest** [comment]
https://asciinema.org/a/lrVznQtvi0aBs7IKuLgjrpyHE

**Pull request merged**